rsc/src/main/java/com/cloudera/livy/rsc/driver/Statement.java

```diff
@@ -36,4 +36,21 @@ public Statement(Integer id, StatementState state, String output) {
   public Statement() {
     this(null, null, null);
   }
+
+  public boolean compareAndTransition(final StatementState from, final StatementState to) {
+    if (state.compareAndSet(from, to)) {
+      StatementState.validate(from, to);
+      return true;
+    }
+    return false;
+  }
+
+  public boolean checkStateAndExecute(final StatementState s, final Runnable runnable) {
```

Comment:
This function creates a pretense that the function is atomic but it is't.
I would rather expose the state directly to Session. The code is shorter too.